File: all-submissions/0020-valid-parentheses/solution.java
class Solution {
    public boolean isValid(String s) {
        if(s==null) return false;
        if(s.length() == 1) return false;
        Stack<Character> st = new Stack<>();
        char[] ch = s.toCharArray();
        for(char c: ch) {
            if(c == '(' || c == '[' || c =='{') {
                st.push(c);
            } else {
                if(st.isEmpty()) return false;
                char curr = st.peek();
                if ((curr == '(' && c == ')') || (curr == '[' && c == ']') || (curr == '{' && c == '}')) {
                    st.pop();
                    continue;
                } else {
                    return false;
                }
            }
        }
        if(st.isEmpty()) return true;
        return false;
        
    }
}

File: all-submissions/0020-valid-parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for char in s:
            if char == '(' or char == '{' or char == '[':
                stack.append(char)
            else:
                if not stack:
                    return False
                ch = stack.pop()
                if (ch == '(' and char == ')') or (ch == '{' and char == '}') or (ch == '[' and char == ']'):
                    continue
                else:
                    return False
        return True if not stack else False
